### ISSUE TITLE
suggestion: entries(observableMap)

### ISSUE THREAD

**Issue by xaviergonz** [mobxjs/mobx#1416]
Would go great alongside the new values(), and keys()

**Comment by mweststrate** [comment]
Was thinking about that. But note that Array.from(observableMap) achieves
the same

Op do 15 mrt. 2018 om 18:34 schreef Javier Gonzalez <
notifications@github.com>:

**Comment by xaviergonz** [comment]
Yep, but I guess entries (as well as its values and keys friends) is an optimized version that doesn't try to create an array if it is already an array?

**Comment by sorenhoyer** [comment]
If there's any performance improvement in entries() as opposed to Array.from(map) then I definitely agree.

**Comment by BruceL33t** [comment]
Just a heads up: https://jsperf.com/set-iterator-vs-foreach/4
We should probably use array spread syntax `[...myMap.values()]` instead of `array.from`, since it seems to be much better optimized for both current and especially the upcoming browsers (look at the huge performance improvement in current bleeding edge chrome 67 canarian - just verified it)

**Comment by samjacobclift** [comment]
Is this not already implemented [here](https://github.com/mobxjs/mobx/blob/8cccd9330a88a2a4dee4c038a60c1fac8ed965d6/src/types/observablemap.ts#L261)? Or am I missing something?

**Comment by quanganhtran** [comment]
Should the issue be resolved already? @xaviergonz

**Comment by mweststrate** [comment]
No, this issue is about a top-level api, not a map api, like these:
https://github.com/mobxjs/mobx/blob/8cccd9330a88a2a4dee4c038a60c1fac8ed965d6/src/api/object-api.ts

Op vr 4 mei 2018 om 10:28 schreef quanganhtran <notifications@github.com>:

**Comment by samjacobclift** [comment]
I see I’ll give the implementation a go!

**Comment by samjacobclift** [comment]
here is the PR 👍 https://github.com/mobxjs/mobx/pull/1536
